File: QueDrop_Android/QueDrop_Driver/app/src/main/java/com/quedrop/driver/utils/Utility.java
package com.quedrop.driver.utils;

import android.app.ActivityManager;
import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.RectF;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.widget.ImageView;
import android.widget.Toast;

import com.bumptech.glide.Glide;

import java.io.File;
import java.util.List;

import okhttp3.MediaType;
import okhttp3.MultipartBody;
import okhttp3.RequestBody;

import static android.content.Context.CONNECTIVITY_SERVICE;

public class Utility {


    public static MultipartBody.Part getFileBody(String key, String filePath) {

        File file = new File(filePath);
        RequestBody fileReqBody = RequestBody.create(MediaType.parse("image/*"), file);
        return MultipartBody.Part.createFormData(key, file.getName(), fileReqBody);

    }

    public static Bitmap createDrawableNode(String strText) {
        Paint mTextPaint = new Paint();
        mTextPaint.setColor(Color.WHITE);
        mTextPaint.setTextAlign(Paint.Align.CENTER);
        mTextPaint.setDither(true);
        mTextPaint.setAntiAlias(true);
        mTextPaint.setFilterBitmap(true);
        mTextPaint.setTextSize(38);
        float vSize = 56;
        Bitmap bitmap = Bitmap.createBitmap((int) vSize, (int) vSize, Bitmap.Config.ARGB_8888);
        Canvas canvas = new Canvas(bitmap);
        Paint paint = new Paint();
        paint.setStyle(Paint.Style.FILL);
        paint.setColor(Color.BLACK);
        paint.setAntiAlias(true);
        RectF rectF = new RectF(0, 0, canvas.getWidth(), canvas.getHeight());

        int cornersRadius = 40;
        canvas.drawRoundRect(rectF, cornersRadius, cornersRadius, paint);
        int xPos = (canvas.getWidth() / 2);
        int yPos = (int) ((canvas.getHeight() / 2) - ((mTextPaint.descent() + mTextPaint.ascent()) / 2));
        canvas.drawText(strText, xPos, yPos, mTextPaint);
        return bitmap;
    }


    public static void toastLong(Context mContext, String message) {

        Toast.makeText(mContext, "" + message, Toast.LENGTH_SHORT).show();
    }

    public static void loadGlideImage(Context context, String path, int placeHolder, ImageView imageView) {
        try {
            if (!path.isEmpty()) {
                Glide.with(context)
                        .load(path)
                        .placeholder(placeHolder)
                        .circleCrop()
                        .into(imageView);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

    }


    public static boolean isNetworkAvailable(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(CONNECTIVITY_SERVICE);
        assert cm != null;
        NetworkInfo networkInfo = cm.getActiveNetworkInfo();
        return networkInfo != null && networkInfo.isConnected();
    }

    public static boolean isAppOnForeground(Context context, String appPackageName) {
        ActivityManager activityManager = (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
        List<ActivityManager.RunningAppProcessInfo> appProcesses = activityManager.getRunningAppProcesses();
        if (appProcesses == null) {
            return false;
        }
        final String packageName = appPackageName;
        for (ActivityManager.RunningAppProcessInfo appProcess : appProcesses) {
            if (appProcess.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND && appProcess.processName.equals(packageName)) {
                //                Log.e("app",appPackageName);
                return true;
            }
        }
        return false;
    }


}

File: QueDrop_Android/QueDrop/app/src/main/java/com/quedrop/customer/utils/Utility.kt
package com.quedrop.customer.utils

import android.content.Context
import android.graphics.*
import android.widget.Toast
import com.google.firebase.messaging.RemoteMessage
import com.google.gson.JsonObject
import com.google.gson.JsonParser
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.MultipartBody
import okhttp3.RequestBody
import java.io.File
import java.text.SimpleDateFormat
import java.util.*


object Utility {
    fun getFileBody(key: String?, filePath: String?): MultipartBody.Part {
        val file = File(filePath)
        val fileReqBody =
            RequestBody.create("image/*".toMediaTypeOrNull(), file)
        return MultipartBody.Part.createFormData(key!!, file.name, fileReqBody)
    }

    fun createDrawableNode(strText: String?): Bitmap {
        val mTextPaint = Paint()
        mTextPaint.color = Color.WHITE
        mTextPaint.textAlign = Paint.Align.CENTER
        mTextPaint.isDither = true
        mTextPaint.isAntiAlias = true
        mTextPaint.isFilterBitmap = true
        mTextPaint.textSize = 38f
        //round? kmnu that
        val vSize = 56f
        val bitmap =
            Bitmap.createBitmap(vSize.toInt(), vSize.toInt(), Bitmap.Config.ARGB_8888)
        val canvas = Canvas(bitmap)
        val paint = Paint()
        paint.style = Paint.Style.FILL
        paint.color = Color.BLACK
        paint.isAntiAlias = true
        val rectF = RectF(0F, 0F, canvas.width.toFloat(), canvas.height.toFloat())
        val cornersRadius = 40
        canvas.drawRoundRect(rectF, cornersRadius.toFloat(), cornersRadius.toFloat(), paint)
        val xPos = canvas.width / 2
        val yPos =
            (canvas.height / 2 - (mTextPaint.descent() + mTextPaint.ascent()) / 2).toInt()
        canvas.drawText(strText!!, xPos.toFloat(), yPos.toFloat(), mTextPaint)
        return bitmap
    }

    fun toastLong(mContext: Context?, message: String) {
        Toast.makeText(mContext, "" + message, Toast.LENGTH_SHORT).show()
    }

    fun convertRemoteMessage(remoteMessage: RemoteMessage): JsonObject {
        val jsonObject = JsonObject() // com.google.gson.JsonObject
        val jsonParser = JsonParser() // com.google.gson.JsonParser
        val map = remoteMessage.data
        var valData: String

        for (mykey in map.keys) {
            valData = map[mykey]!!
            try {
                jsonObject.add(mykey, jsonParser.parse(valData))
            } catch (e: Exception) {
                jsonObject.addProperty(mykey, valData)
            }

        }
        return jsonObject
    }
    fun convertUTCDateToLocalDate(date: String,format:String): String {
        val dateStr = date
        val df = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault())
        df.timeZone = TimeZone.getTimeZone("UTC")
        val date = df.parse(dateStr)
        df.timeZone = TimeZone.getDefault()

        val localDate = df.format(date)

        val formatter = SimpleDateFormat(format, Locale.getDefault())
        return formatter.format(df.parse(localDate))
    }

}
